src/Utils/kernel_launching.jl

```diff
@@ -193,43 +215,66 @@ If the worksize is 0 after adjusting for active cells, the function returns `not
 - `reduced_dimensions`: A tuple specifying the dimensions to be reduced in the work distribution. Default is an empty tuple.
 - `location`: The location of the kernel execution, needed for `include_right_boundaries`. Default is `nothing`.
 - `active_cells_map`: A map indicating the active cells in the grid. If the map is not a nothing, the workspec will be disregarded and 
-                     the kernel is configured as a linear kernel with a worksize equal to the length of the active cell map. Default is `nothing`.
+                      the kernel is configured as a linear kernel with a worksize equal to the length of the active cell map. Default is `nothing`.
 """
+@inline function configure_kernel(arch, grid, workspec, kernel!;
+                                  exclude_periphery = false,
+                                  reduced_dimensions = (),
+                                  location = nothing,
+                                  active_cells_map = nothing)
+
+    if !isnothing(active_cells_map) # everything else is irrelevant
+        workgroup = min(length(active_cells_map), 256)
+        worksize = length(active_cells_map)
```

Comment:
we probably need to keep this here for non-active domains
```suggestion
        if worksize == 0
            return nothing
        end
```

Reply:
ok nothing, I see it has moved in the `launch!`